### ISSUE TITLE
Add mailbox name printf(3)-like sequence to $status_format.

### ISSUE THREAD

**Issue by hsanson** [neomutt/neomutt#1354]
## What would you like to be able to do?

The status_format supports the `%f` sequence that prints the full pathname of the current mailbox. 
With virtual folders this sequence shows the whole search query like:

```
notmuch:///home/Mail?type=threads&query=path%3Aaccount1%2F** AND tag%3Ainbox AND NOT tag%3Akilled AND NOT tag%3A :: C: 30 F: 3 S: 22M
```

This makes it hard to see what is the current mailbox. Instead it should show the name of the virtual folder:

```
Inbox :: C: 30 F: 3 S: 22M
```

## What needs to change in NeoMutt?

When declaring a virtual mailbox with:

```
virtual-mailboxes "Inbox"   "notmuch://?query=path:account1/** AND tag:inbox AND NOT tag:killed AND NOT tag:spam"
```

Modify the `%f` sequence in $status_format to show `Inbox` instead of the whole notmuch query. Or better create a new sequence that shows the virtual mailbox name instead of the query.

## Have you tried any alternative methods?

Read docs and could not find any sequence that shows the mailbox name rather than the full path or query.

## Additional Info

I am not 100% sure but I think before neomutt was showing the mailbox name instead of the full path or query. At some point after upgrading neomutt it started showing only the search query.

**Comment by Austin-Ray** [comment]
Thank you for your bug report. I can confirm that this is an issue! Offending commit is 5fa35b404.

I've built off `master` daily for the past two months and haven't noticed! Thanks!

**Comment by Austin-Ray** [comment]
@flatcap How should we handle this issue?

According to the documentation:
> %f - The full pathname of the current mailbox

5fa35b4 makes the `%f` consistent across the various mailbox backends, but `notmuch` users don't get nice looking pathnames like `imap` or `maildir` users, especially with very complex queries. It would be nice to use the description, but there is no format option to do so.

As an aside, why can't we set descriptions for normal mailboxes?

**Comment by flatcap** [comment]
Currently, the sidebar uses the description if it's set.
I think the status bar should work the same way for now.

> As an aside, why can't we set descriptions for normal mailboxes?

The `desc`ription was only added by the Notmuch patch, so there isn't much code behind it (yet).
Once before, I've mocked up a command which takes two parameters:
- `named-mailboxes DESC MAILBOX`

(hmm... I can't find the source, but it was only a few lines to implement)

When we add `named-mailboxes` we could add a dedicated expando for the description.

---

I think I've changed my mind (whilst writing this).
The above means the behaviour of `%f` would change...

New plan:  Implement `%D` (description) for `status_format` and `sidebar_format`.
(they default to the full mailbox name if the description is missing)

Then add a `named-mailboxes` command.

**Comment by Austin-Ray** [comment]
@flatcap Looking into this further, `%f` in the case of `virtual-mailboxes` is supposed to use the description! However, somewhere along the line, we lose track of it.

https://github.com/neomutt/neomutt/blob/b71ac2bae5d471b2a32452d9103d4164f27f729f/status.c#L127-L131
Take a look at this segment of code from the `%f` case. It tries to use the `desc` variable, but if you add a `mutt_debug(1, "Description: %s\n", m->desc);` above the conditional, you'll get `NULL` in the logs! No wonder we fallback to the path.

Before we can implement `%D`, we need to find where we lose `desc` since that would affect future named mailboxes as well.

**Comment by flatcap** [comment]
/me bisects quickly...
Surprise, surprise!  I've broken it, with commit: 5fa35b40439a9fc9cacfdeff0c135083603c6bab refactor/simplify status %f
I factored out a function that was only used in once place.
I'll fix it later this evening.

**Comment by Austin-Ray** [comment]
> Surprise, surprise! I've broken it, with commit: 5fa35b4 refactor/simplify status %f

If I had known you were going to `git bisect`, I would have saved you the effort. I had reported the offending commit higher up.

I think reverting that code will hide the true cause of the issue. When the `virtual-mailboxes` declaractions are parsed, we get the correct `desc`, which the sidebar uses. However, when we get to the status bar, we've lost the `desc`. How? More importantly, where? And how come the sidebar is able to continue using the `desc` after we lost it? Are the duplicating data across modules?

**Comment by Austin-Ray** [comment]
Fixed in 3a5a978dc9cbd5f1dbbeaa7b298a01f7245bdd31
